src/mpi/info/infoutil.c

```diff
@@ -67,10 +67,6 @@ void MPIR_Info_setup_env(MPIR_Info * info_ptr)
     info_init(info_ptr);
     /* FIXME: Currently this info object is missing some data as
      * defined by the standard. */
-
-    if (MPIR_Process.memory_alloc_kinds != NULL) {
-        MPIR_Info_push(info_ptr, "mpi_memory_alloc_kinds", MPIR_Process.memory_alloc_kinds);
-    }
```

Comment:
The MPI Forum discussion yesterday settled on the value of this key in MPI_INFO_ENV should be the requested value to align with all the other predefined keys. To me, that also means reporting anything for this key is optional, and others agreed.

It did not seem that anyone was in favor of delaying the 4.1 ratification to fix this issue, since no implementation exists and there are no users. Users of the world model who wish to know the supported memory kinds will be advised to query a predefined MPI object such as MPI_COMM_WORLD or MPI_COMM_SELF.

I'll update the PR with a note in the CHANGELOG so this context is captured. I believe what we have will be aligned with the text shortly.